GoogleDataTransport/GDTCORLibrary/GDTCORMetrics.m

```diff
@@ -54,6 +54,8 @@ - (instancetype)initWithCollectionStartDate:(NSDate *)collectionStartDate
 
 + (instancetype)metricsWithMetricsMetadata:(GDTCORMetricsMetadata *)metricsMetadata
                            storageMetadata:(GDTCORStorageMetadata *)storageMetadata {
+  NSParameterAssert(metricsMetadata);
```

Comment:
Why not make the parameters nonnull instead?

Reply:
Even with the analyzer?

Reply:
Interesting– so the he analyzer catches the issue in the simple examples I have above but it's unable to see the risks with passing the `snapshottedMetricsMetadata` from line 99 into the return on 126 here: 

https://github.com/google/GoogleDataTransport/blob/b12ea6cdb22c6205dacbe88cf9e60f3398e73429/GoogleDataTransport/GDTCORLibrary/GDTCORMetricsController.m#L98-L129

Reply:
Removed the asserts in 5c5c9ad